services/mfa/mfa.go

```diff
@@ -51,6 +54,69 @@ func getIssuerFromUrl(uri string) string {
 	return url.QueryEscape(issuer)
 }
 
+func generateCode(n int) string {
+	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = charset[rand.Int63()%int64(len(charset))]
```

Comment:
For the code generation, I'd use `crypto/rand` instead of `math/rand` since you want a cryptographically random function for this security function.